workflow/kfdrc_production_amplicon_architect.cwl

```diff
@@ -0,0 +1,286 @@
+      \ to use", default: "GRCh38"}
+  tumor_align_file: {type: File, doc: "Tumor read alignment file. Can be cram or bam",
+    secondaryFiles: ['^.bai?', '.crai?']}
+  output_basename: {type: string, doc: "File name prefix for steps that require it"}
+  mosek_license_file: {type: File, doc: "This tool uses some software that requires\
+      \ a license file. You can get a personal or institutional one from https://www.mosek.com/license/request/."}
+  # If cram input and cnvkit not run
+  reference: {type: 'File?', doc: "fasta file, needed if cram input or cnv kit cnn\
+      \ not already built", secondaryFiles: [.fai]}
+  # If proper .cns file exists
+  cnvkit_cns: {type: 'File?', doc: ".cns file from previous CNVkit run, if available.\
+      \ DO NOT USE .call.cns"}
+  # If CNVkit needs running and .cnn file exists
+  cnvkit_cnn: {type: 'File?', doc: "If running using an existing .cnn, supply here"}
+  # If CNVkit needs running and .cnn NOT available
+  normal_align_file: {type: 'File?', doc: "Normal read alignment file. Can cram or\
+      \ bam", secondaryFiles: [{pattern: '^.bai', required: false}, {pattern: '.crai',
+        required: false}]}
+  annotation_file: {type: 'File?', doc: "refFlat.txt file, needed if cnv kit cnn not\
+      \ already built"}
+  male_input_flag: {type: 'boolean?', doc: "Is the input male?", default: false}
+
+outputs:
+  aa_cnv_seeds: {type: File, doc: "Bed file with candidate regions to search", outputSource: prepare_aa/aa_cnv_seeds}
+  aa_summary: {type: File, doc: "summary for all amplicons detected by AA", outputSource: amplicon_architect/summary}
+  aa_cycles: {type: 'File[]', doc: "text file for each amplicon listing the edges\
+      \ in the breakpoint graph, their categorization (sequence, discordant, concordant,\
+      \ source) and their copy counts", outputSource: amplicon_architect/cycles}
+  graph: {type: 'File[]', doc: 'A text file for each amplicon listing the edges in
+      the breakpoint graph, their categorization (sequence, discordant, concordant,
+      source) and their copy counts', outputSource: amplicon_architect/graph}
+  sv_pdf: {type: 'File[]', doc: "PDF image file displaying the SV view of AA", outputSource: amplicon_architect/sv_pdf}
+  sv_png: {type: 'File[]', doc: "PNG image file displaying the SV view of AA", outputSource: amplicon_architect/sv_png}
+  amplicon_classification_profiles: {type: 'File[]?', doc: "abstract classification\
+      \ of the amplicon", outputSource: amplicon_classifier/amplicon_classification_profiles}
+  gene_list: {type: 'File[]?', doc: "genes present on amplicons with each classification",
+    outputSource: amplicon_classifier/gene_list}
+  ecDNA_cts: {type: 'File[]?', doc: "not yet defined", outputSource: amplicon_classifier/ecDNA_cts}
+
+steps:
+  samtools_tumor_cram_to_bam:
+    run: ../tools/samtools_calmd.cwl
+    when: $(inputs.input_reads.nameext == ".cram")
```

Comment:
Should this also check that input_reads != null?

Reply:
It's a required input. It's checked on input.

Reply:
Yup, that

Reply:
Ok, then it should be removed in the step after this one.

Reply:
No it shouldn't the normal_align_file is optional. You need to check if it's there.